src/sentry/api/helpers/group_index/update.py

```diff
@@ -836,6 +836,9 @@ def handle_assigned_to(
     )
     if assigned_actor:
         for group in group_list:
+            if features.has("organizations:participants-purge", group.organization):
+                GroupAssignee.objects.deassign(group, acting_user)
```

Comment:
Passing through `deassign` before `assign` (when reassigning) neatly sends the unassignment notification